dace/sdfg/validation.py

```diff
@@ -191,6 +191,18 @@ def validate_state(state: 'dace.sdfg.SDFGState',
         except Exception as ex:
             raise InvalidSDFGNodeError("Node validation failed: " + str(ex), sdfg, state_id, nid) from ex
 
+        if Config.get_bool('check_race_conditions'):
+            if isinstance(node, nd.AccessNode):
+                incoming_edge_memlet_ranges = []
+                for e in state.in_edges(node):
+                    incoming_edge_memlet_ranges.append(e.data.subset)
```

Comment:
you'd probably want to use `e.data.dst_subset` here, in case it's an access node to access node edge.